src/lib/cache/CategoryLookupTablesCache.js

```diff
@@ -60,8 +85,54 @@ export default class CategoryLookupTablesCache {
     const countryCode = localeString.substr(0, 2);
     this.lookupTablesCache.set(countryCode, Promise.resolve(rawCategoryLists));
   }
+
+  /**
+   * Updates a valid entry in the ttl cache. Leaves resolved promise in the cache as
+   * long as possible to prevent slow responses in between.
+   *
+   * This introduces a potential resource waste/timing issue
+   *  T0: req1 - cache still valid, background reloading triggers
+   *  T1: req2 - cache invalid, new request started, not reusing the promise of req1
+   *  T2: res2 - updates cache
+   *  T3: res1 - updates cache again (with potentially older result)
+   *
+   * A solution could be storing the reload promise with the ttl-cache entry and using it
+   * when the ttl entry is expired. This would increase the complexity too much for
+   * an edge case that no one will care about.
+   */
+  reloadInBackground(countryCode: string) {
```

Comment:
+0: I dislike this code duplication, but I think I understand the reason behind it.

Reply:
Me too, imho we could easily have the category cache extend urldata cache as well and get rid of the duplication. Will add to the cleanup backlog